libbeat/docs/shared-ssl-config.asciidoc

```diff
@@ -87,25 +87,154 @@ Example module with SSL enabled:
 ----
 endif::[]
 
-[float]
-=== Configuration options
+[discrete]
+[[ssl-common-config]]
+=== Common configuration options
 
-You can specify the following options in the `ssl` section of the +{beatname_lc}.yml+ config file:
+In the +{beatname_lc}.yml+ config file, you can specify the following common options in the
+`ssl` section of each output that supports SSL.
 
 [float]
+[[enabled]]
 ==== `enabled`
 
-The `enabled` setting can be used to disable the ssl configuration by setting
-it to `false`. The default value is `true`.
+To disable the ssl configuration, set the value to `false`. The default value is `true`.
 
-NOTE: SSL settings are disabled if either `enabled` is set to `false` or the
+[NOTE]
+=====
+SSL settings are disabled if either `enabled` is set to `false` or the
 `ssl` section is missing.
+=====
+
+[float]
+[[key-passphrase]]
+==== `key_passphrase`
+
+The passphrase used to decrypt an encrypted key stored in the configured `key` file.
+
+[float]
+[[supported-protocols]]
+==== `supported_protocols`
+
+List of allowed SSL/TLS versions. If SSL/TLS server decides for protocol versions
+not configured, the connection will be dropped during or after the handshake. The
+setting is a list of allowed protocol versions:
+`SSLv3`, `TLSv1` for TLS version 1.0, `TLSv1.0`, `TLSv1.1`, `TLSv1.2`, and
+`TLSv1.3`.
+
+The default value is `[TLSv1.1, TLSv1.2, TLSv1.3]`.
+
+[float]
+[[cipher-suites]]
+==== `cipher_suites`
+
+The list of cipher suites to use. The first entry has the highest priority.
+If this option is omitted, the Go crypto library's https://golang.org/pkg/crypto/tls/[default suites]
+are used (recommended). Note that TLS 1.3 cipher suites are not
+individually configurable in Go, so they are not included in this list.
+
+// tag::cipher_suites[]
+The following cipher suites are available:
+
+[options="header"]
+|===
+| Cypher | Notes
+| ECDHE-ECDSA-AES-128-CBC-SHA |
+| ECDHE-ECDSA-AES-128-CBC-SHA256 | TLS 1.2 only. Disabled by default.
+| ECDHE-ECDSA-AES-128-GCM-SHA256 | TLS 1.2 only.
+| ECDHE-ECDSA-AES-256-CBC-SHA |
+| ECDHE-ECDSA-AES-256-GCM-SHA384 | TLS 1.2 only.
+| ECDHE-ECDSA-CHACHA20-POLY1305 | TLS 1.2 only.
+| ECDHE-ECDSA-RC4-128-SHA | Disabled by default. RC4 not recommended.
+| ECDHE-RSA-3DES-CBC3-SHA |
+| ECDHE-RSA-AES-128-CBC-SHA |
+| ECDHE-RSA-AES-128-CBC-SHA256 | TLS 1.2 only. Disabled by default.
+| ECDHE-RSA-AES-128-GCM-SHA256 | TLS 1.2 only.
+| ECDHE-RSA-AES-256-CBC-SHA |
+| ECDHE-RSA-AES-256-GCM-SHA384 | TLS 1.2 only.
+| ECDHE-RSA-CHACHA20-POLY1205 | TLS 1.2 only.
+| ECDHE-RSA-RC4-128-SHA | Disabled by default. RC4 not recommended.
+| RSA-3DES-CBC3-SHA |
+| RSA-AES-128-CBC-SHA |
+| RSA-AES-128-CBC-SHA256 | TLS 1.2 only. Disabled by default.
+| RSA-AES-128-GCM-SHA256 | TLS 1.2 only.
+| RSA-AES-256-CBC-SHA |
+| RSA-AES-256-GCM-SHA384 | TLS 1.2 only.
+| RSA-RC4-128-SHA | Disabled by default. RC4 not recommended.
```

Comment:
@kvch @faec I'm afraid we need to clean up the list a little (in code and documentation)